test/e2e/kubelet_stats.go

```diff
@@ -0,0 +1,173 @@
+
+// ReadKubeletMetrics reads metrics from the kubelet server running on the given node
+func ReadKubeletMetrics(c *client.Client, nodeName string) ([]KubeletMetric, error) {
+	body, err := getKubeletMetrics(c, nodeName)
+	if err != nil {
+		return nil, err
+	}
+
+	metric := make([]KubeletMetric, 0)
+	for _, line := range strings.Split(string(body), "\n") {
+
+		// TODO: String parsing is such a hack, but getting our rest client/proxy to cooperate with prometheus
+		// client is weird, we should eventually invest some time in doing this the right way.
+		if !strings.HasPrefix(line, fmt.Sprintf("%v_", metrics.KubeletSubsystem)) {
+			continue
+		}
+		keyVal := strings.Split(line, " ")
+		if len(keyVal) != 2 {
+			return nil, fmt.Errorf("Error parsing metric %q", line)
+		}
+		keyElems := strings.Split(line, "\"")
+
+		latency, err := strconv.ParseFloat(keyVal[1], 64)
+		if err != nil {
+			continue
+		}
+
+		methodLine := strings.Split(keyElems[0], "{")
+		methodList := strings.Split(methodLine[0], "_")
+		if len(methodLine) != 2 || len(methodList) == 1 {
+			continue
+		}
+		method := strings.Join(methodList[1:], "_")
+
+		var operation, rawQuantile string
+		var quantile float64
+
+		switch method {
+		case metrics.PodWorkerKey:
+			// eg: kubelet_pod_worker_latency_microseconds{operation_type="create",pod_name="foopause3_default",quantile="0.99"} 1344
+			if len(keyElems) != 7 {
+				continue
+			}
+			operation = keyElems[1]
+			rawQuantile = keyElems[5]
+			break
+
+		case metrics.SyncPodsKey:
+			// eg:  kubelet_sync_pods_latency_microseconds{quantile="0.5"} 9949
+			fallthrough
+
+		case metrics.PodStartKey:
+			// eg: kubelet_pod_start_latency_microseconds{quantile="0.5"} 123
+			fallthrough
+
+		case metrics.PodStatusKey:
+			// eg: kubelet_generate_pod_status_latency_microseconds{quantile="0.5"} 12715
+			if len(keyElems) != 3 {
+				continue
+			}
+			operation = ""
+			rawQuantile = keyElems[1]
+			break
+
+		case metrics.ContainerManagerOperationsKey:
+			// eg: kubelet_container_manager_latency_microseconds{operation_type="SyncPod",quantile="0.5"} 6705
+			fallthrough
+
+		case metrics.DockerOperationsKey:
+			// eg: kubelet_docker_operations_latency_microseconds{operation_type="info",quantile="0.5"} 31590
+			if len(keyElems) != 5 {
+				continue
+			}
+			operation = keyElems[1]
+			rawQuantile = keyElems[3]
+			break
+
+		case metrics.DockerErrorsKey:
+			Logf("ERROR %v", line)
+
+		default:
+			continue
+		}
+		quantile, err = strconv.ParseFloat(rawQuantile, 64)
+		if err != nil {
+			continue
+		}
+		metric = append(metric, KubeletMetric{operation, method, quantile, time.Duration(int64(latency)) * time.Microsecond})
+	}
+	return metric, nil
+}
+
+// HighLatencyKubeletOperations logs and counts the high latency metrics exported by the kubelet server via /metrics.
+func HighLatencyKubeletOperations(c *client.Client, threshold time.Duration, nodeName string) (int, error) {
```

Comment:
I'd rather have it output slice of "bad metrics" and allow user to decide what to do with them (i.e. she might want to additionally get events for bad X).